tutorials/plot_sleep.py

```diff
@@ -20,6 +20,46 @@
 
 raw = mne.io.read_raw_edf(psg_fname, stim_channel=False)
 annotations = mne.read_annotations(hyp_fname)
+
+##############################################################################
+# preprocessing annotations
+
+# 1step: resample annotation in 30s annotations
+annot = pd.DataFrame()
+annot["onset"] = annotations.onset
+annot["description"] = annotations.description
+annot["duration"] = annotations.duration
+
+# add temporarily a last event to have a correct resampling
+last_onset = annot.onset.values[-1]
+last_duration = annot.duration.values[-1]
+annot.loc[annot.shape[0]] = [last_onset + last_duration, "end", 0]
+
+annot = annot.set_index('onset')
+annot.index = pd.to_timedelta(annot.index, unit='s')
+annot = annot.resample('30s').ffill()
+annot.reset_index(inplace=True)
+annot.onset = annot.onset.dt.total_seconds()
+annot["duration"] = 30.
```

Comment:
I see it now. That's a bit long but it's exactly what I had in mind. Maybe just make it a function so we know where
pandas kungfu ends.